libpod/runtime_img.go

```diff
@@ -1306,51 +1217,40 @@ func ImageDangling() ImageResultFilter {
 // ImageLabel allows you to filter by images labels key and/or value
 func ImageLabel(labelfilter string) ImageResultFilter {
 	// We need to handle both label=key and label=key=value
-	return func(i inspect.ImageResult) bool {
+	return func(i *image.Image) bool {
 		var value string
 		splitFilter := strings.Split(labelfilter, "=")
 		key := splitFilter[0]
 		if len(splitFilter) > 1 {
 			value = splitFilter[1]
 		}
-		for labelKey, labelValue := range i.Labels {
-			// handles label=key
-			if key == labelKey && len(strings.TrimSpace(value)) == 0 {
-				return true
-			}
-			//handles label=key=value
-			if key == labelKey && value == labelValue {
-				return true
-			}
+		labels, err := i.Labels()
+		if err != nil {
+			return false
+		}
+		if len(strings.TrimSpace(labels[key])) > 0 && len(strings.TrimSpace(value)) == 0 {
+			return true
+		}
+		if labels[key] == value {
```

Comment:
Cleaner to do 

return labels[key] == value

Reply:
fixed